src/util.h

```diff
@@ -80,8 +81,21 @@ static const int64_t CENT = 1000000;
 void MilliSleep(int64_t n);
 
 extern int GetDayOfYear(int64_t timestamp);
-extern std::map<std::string, std::string> mapArgs;
-extern std::map<std::string, std::vector<std::string> > mapMultiArgs;
+
+/**
+ * Allows search of mapArgs and mapMultiArgs in a case insensitive way
+ */
+
+struct mapArgscomp
+{
+   bool operator() (const std::string& lhs, const std::string& rhs) const
+   {
+       return strcasecmp(lhs.c_str(), rhs.c_str()) < 0;
+   }
+};
+
+extern std::map<std::string, std::string, mapArgscomp> mapArgs;
+extern std::map<std::string, std::vector<std::string>, mapArgscomp> mapMultiArgs;
```

Comment:
Can you make a type alias for `map<string, string, mapArgscomp>`?
Then you can use the alias in all the declarations.

Reply:
absolutely let me come up with a good typedef name for it and test it out make sure it all falls into place